### ISSUE TITLE
plexdrive locks randomly without reason

### ISSUE THREAD

**Issue by kuarthepirat** [dweidenfeld/plexdrive#261]
Hello,

I am running the 5.0.0 release on Arch Linux with the 4.9.61-1-lts kernel. Every 24-48 h my plexdrive becomes unresponsive and I can't access the files. At least that's what it seems like, because in the logs for plexdrive it seems to be getting the files content.

I have attached some of the logs for getting my simple check file here: https://pastebin.com/bGnJv6bh

Any idea what to look for?

Kind Regards

**Comment by AndreAuth** [comment]
I have got the same behavior. Only a pkill , fusermount and remount solves this. It has run also for 2 month on my Server. Looking fortward for a Solution . I thought i am the only one

**Comment by kuarthepirat** [comment]
That is exactly what I am experiencing. I tried deleting the cache file, downgrading the kernel, downgrading plex... none have worked so far.

**Comment by AndreAuth** [comment]
Me too. Have you tried another Version of plexdrive so far?

**Comment by kuarthepirat** [comment]
It has been 2 days so far and my system is running stable again. I compiled the latest master from github on my machine and am using that binary.

**Comment by rafuz** [comment]
I am on the same boat.
It worked for 4 months flawlessly until august. Then it started to hang without reason, nothing showing in logs. It simply stops to download chunks when accessing files. It seems completely random, sometimes I can go on a week, sometimes only few hours. Tried to go back to prescribe 4.10 but it hangs the same way.

Someone has any ideas?

**Comment by kuarthepirat** [comment]
Unfortunately you are right... I was just lucky reaching almost 3 days this time. Compiling the master does not do anything :-)

**Comment by AndreAuth** [comment]
Me too. Still crashing almost every day between 6:30 PM to 11:00 PM.

After the pkill its working again.

There is nothing in the Logs. It stops working. 

Does anyone have a script to check if plexdrive is working or not?

**Comment by kuarthepirat** [comment]
I am using Arch Linux, LTS Kernel on a Dedicated Server on Online.net. Plexdrive is started via systemd and I have yet to find a script to check if it is working or not. My first idea was to access a file with "timeout 5 cat /mnt/gdrive/.check" but even the timeout is locked until the plexdrive is restarted so that is not working.

**Comment by AndreAuth** [comment]
Maybe we could look in the Log Files. 

If the automatic refresh interval hasn't been fired for maybe 2-3 Minutes we could automaticly force the restart.

I know this is a work around but i dont know the GO Language to find and fix the Problem. Maybe dweidenfeld could say something for this.

**Comment by rafuz** [comment]
I think that is better to try to find the root cause of this problem instead of using a workaraund (AKA: restart plexdrive)
At least we know that is impacting more than one system with different OS.
I need to know the command line options both of you use for starting plexdrive, the OS version and the hardware specs of the cpu of your server.
I think that the problem is something related to multi-threading and IO but I need more info.

**Comment by AndreAuth** [comment]
Here are my specs and command:

Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                8
On-line CPU(s) list:   0-7
Thread(s) per core:    1
Core(s) per socket:    8
Socket(s):             1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 77
Model name:            Intel(R) Atom(TM) CPU  C2750  @ 2.40GHz
Stepping:              8
CPU MHz:               2393.905
BogoMIPS:              4787.81
Virtualization:        VT-x


Debian GNU/Linux 8

32864616 K total memory

mount --uid=1001 --gid=1001 --max-chunks=400 -o allow_other -v 2

**Comment by kuarthepirat** [comment]
I am on a Intel(R) Xeon(R) CPU E3-1240 v3 @ 3.40GHz with 32 GB of ECC RAM. My physical storage are 2 480 GB SSDs in a Software RAID1.

plexdrive-current mount --max-chunks=200 -o allow_other --refresh-interval=30s --cache-file=/home/kuar/.plexdrive/gdrive-1/cache.bolt -c /home/kuar/.plexdrive/gdrive-1 /mnt/gdrive-1

**Comment by SeppGeneral** [comment]
CPU: | Intel Xeon E5-1650 3.2Ghz + ILO
RAM: | 32GB
Hard Drive(s): | 2x 600GB SAS 15K HW RAID | RAID1
Bandwidth: | Unltd @ 1Gbps

plexdrive-linux-amd64 -o allow_other,read_only --clear-chunk-max-size=50G --clear-chunk-age=12h /mnt/NAS

**Comment by rafuz** [comment]
We all have different CPUs and different memory configurations. I am trying to mount plexdrive with the "vanilla" settings used by the developer too:

`/usr/bin/plexdrive5 mount /home/plex/gdrive/ --cache-file=/home/plex/plexdrivecache.bolt -v 3`

I only relocate the bolt cache.
I am monitoring the process so I hope to find something strange when it locks.

I have this CPU:
model name: Intel(R) Xeon(R) CPU E3-1225 V2 @ 3.20GHz
16 GB RAM

**Comment by AndreAuth** [comment]
Crashed again a few Minutes ago . Nothing in the Logs.

**Comment by rafuz** [comment]
I am monitoring it but still no hangs.

Before I was using different options to start it:
`plexdrive5 mount $MPOINT --chunk-check-threads=8 --chunk-load-ahead=4 --chunk-load-threads=8 --chunk-size=25M --max-chunks=10 --refresh-interval=1m --fuse-options=allow_other,read_only --cache-file=/home/plex/plexdrivecache.bolt -v 3`

I think that at least in my scenario the number for --chunk-check-threads and --chunk-load-threads could be the culprit. I have to wait to see what happens without this parameters.

@AndreAuth: Have you tried to lower or completely remove --max-chunks? When I tryied to use it on my server it simply slowers the time Plex needs to open new files over time.

**Comment by AndreAuth** [comment]
ok i will try them and i will comment if a crash happens. Otherwise i will wait until tomorrow

**Comment by doul59** [comment]
Hi,  I have the same problem :
I'm on v3 since 6 months : 0 problems
but since 2 weeks, some troubles appear with Google drive (gsuite)
V3 or V5 same scenario with plexdrive/plex

**Comment by AndreAuth** [comment]
Crashed again 😒

**Comment by kuarthepirat** [comment]
Just out of curiosity... which packages did you install on your system in the last weeks?

**Comment by AndreAuth** [comment]
nothing new. Maybe a plex update. Nothing more.

**Comment by kuarthepirat** [comment]
grep installed /var/log/dpkg.log should give you a list. Maybe we can find the software that interferes with plexdrive.

**Comment by AndreAuth** [comment]
Here you go:

https://pastebin.com/9spG4gFx

But it happened already before the first log entry.

I have the first crash on the 29th October. Since then almost every day between 7PM To 11PM

**Comment by rafuz** [comment]
@kuarthepirat: no new packages installed from the time the issue has come out

It was hung right now. I had to kill the process and force unmount.
No signs of hang in the logs

**Comment by AndreAuth** [comment]
Maybe there is a Connection Problem to Gdrive . I dont know the Code but: is there any timeout or Reconnect mechanism ? Is the Download of the Chucks synchrous and plexdrive waits for the Download and freezes? This could be a possible explanation .

**Comment by rafuz** [comment]
At least for me there is no sign of connections lost. I have other services and when it happens I am often connected in ssh and nothing seems to disconnect.

**Comment by AndreAuth** [comment]
Yeah here too. But i mean s Connection Problem to Google

**Comment by fraviktor** [comment]
Same problem here with 2 different servers with the following settings:
Google Drive / GSuite drive for both & PLEX
Plexdrive service with version V4 (the problems started with the Plexdrive V5 and I thought it will change with the downgrade to v4)
**1st server**
Intel Atom C2750 (8 core)
16 GB RAM
500 Mbps
Mount
plexdrive-linux-amd64-v4 -c /home/user/.plexdrive --fuse-options=allow_other,read_only --refresh-interval=1m /home/user/GDrive

**2nd server**
Intel(R) Xeon(R) CPU E5-1650 0 @ 3.20GHz (4 core)
5.5 GB RAM
1 Gbps
plexdrive-linux-amd64-v4 -v2 -c /home/user/.plexdrive -o allow_other,read_only /home/user/GSuite

**Comment by rafuz** [comment]
It hangs with v4 too for me.
It is really a strange behavior, maybe @dweidenfeld could chime in and give us some advice regarding this topic?
It is really annoying right now.

**Comment by AndreAuth** [comment]
I know this is the Plexdrive Github but for the Workarround until it works stable again:

Have anyone tried the new version of Rclone?

https://forum.rclone.org/t/new-feature-cache/4259

**Comment by dweidenfeld** [comment]
Well there is no timeout right now. I am running out for months without a problem. Maybe you could check your network connection. I could imagine that your connection drops for this time while Plex or something else is accessing the drive.

**Comment by rafuz** [comment]
I have a monitor on gdrive api on monit and it never goes down.
My main server is hoster in OVH RBX datacenter, the others experiencing the same issue are using different hosters?
Can we have a way to get more informations when this issue occurs?

**Comment by AndreAuth** [comment]
My hoster is Scaleway

**Comment by rafuz** [comment]
> Have anyone tried the new version of Rclone?

Every time I use rclone mount I get a ban from gdrive. No way for me, plexdrive was (and still should be) much better.

**Comment by fraviktor** [comment]
I forgot to mention that my gdrive data is encrypted and I use rclone mount crypt to decrypt the data from plexdrive mount. Can it be possible that the issues are related to rclone only? I need to check this out next time when it goes down.

**Comment by kuarthepirat** [comment]
My gdrive data is not encrypted, yet I still have the problems.

**Comment by djsecrist** [comment]
> Can it be possible that the issues are related to rclone only?

Not likely...  I use rclone mount crypt, and Plexdrive has been rock-solid for me for months! I am not experiencing the issues that you guys are describing.  Sorry...

I'm interested in the new caching of rclone though.  The write capability of rclone is something that I am missing using PlexDrive. rufuz, were you using the caching feature when you were banned?

**Comment by AndreAuth** [comment]
My Data is encrypted but via Encfs so plexdrive > Encfs. So no rclone in my configuration

**Comment by AndreAuth** [comment]
Another freeze about 5 Minutes ago . The Server is still available. No connection drop

**Comment by rafuz** [comment]
Same here, got a freeze few minutes ago.
No connections drop to google api.

I don't use encryption, just plexdrive to mount gdrive, my media folder is in the root.
I mount plexdrive as non-root user.

**Comment by AndreAuth** [comment]
> I mount plexdrive as non-root user.

Me too.

**Comment by kuarthepirat** [comment]
I am now pretty sure, that this issue is related to another software package on our boxes. At first I tried to rollback the kernel to a version I knew was working - with the effect of a lockup 5 h later.

I decided to rollback my whole system back to a state I knew should work. So I downgraded all official ArchLinux packages to their state on 15th October. And guess what, I didn't have a lockup since 72 h (it usually locks down in the evening at around 8 pm). It is probably too soon, to be 100 % sure, but I am pretty confident, this issue is related to a software package or security upgrade we all have installed on our boxes, that is not on other boxes.

The downgrade did not affect software packages that are not in the official ArchLinux repository. That means my PMS is running on 1.9.7

**Comment by AndreAuth** [comment]
But we all have different Operation Systems. My Plex Version is 1.9.7.4460 . I dont know that i have made an update on the day the first freeze happened.

**Comment by kuarthepirat** [comment]
Yes, we do, but there must be something we changed on our systems. It's either the connectivity to google that was changed by our host (I can rule that out, because of my downgrade), plexdrive itself, which hasn't changed for month, a defect on our boxes (I can rule that out too) and other software on the system affecting plexdrive.

**Comment by rafuz** [comment]
For sure is something related to software packages or OS. 
We are using different datacenters and different hardware.
@kuarthepirat are you able to make a list of packages that you downgraded or removed from your system?
Anyway after 72h I would not be sure that it does not happens anymore.

**Comment by kuarthepirat** [comment]
Yeah, 72h is not enough, I will be sure after a week or something like that. The total package list is on: https://pastebin.com/y8Ek0zcN unfortunately I am running ArchLinux with its rolling releases so there are a lot of packages.

**Comment by rafuz** [comment]
I will try to compare with my software updates from the end of august.
Anyway the issue came out when I was using ubuntu 16.10 and was already out of support. So I didn't installed any updates until October when I did a distribution upgrade.

A list of software I am also running:
UnionFS
Mono for Sonarr/Radarr/Jackett
Deluge
Sabnzbd
Monit
Syncthing
Rclone
Plex Media Server

Could be an option try to put plexdrive in Docker?

**Comment by dweidenfeld** [comment]
There is no good way of running plexdrive in an idiomatic way in a docker container. 

I am already planning a new version of plexdrive or maybe a parallel program for running plexdrive in a docker container with failover scenario.

But as always... Time is limited.

**Comment by rafuz** [comment]
Important question... anyone using Ombi or jd2 headless on your systems?

**Comment by AndreAuth** [comment]
No

**Comment by kuarthepirat** [comment]
No Ombi and no JD on my system. Just a few docker containers with sonarr, radarr and traefik.

**Comment by rafuz** [comment]
@kuarthepirat Are you still running stable?

I have found that the before the first freeze I installed this packages:

`Start-Date: 2017-08-27  18:53:44
Commandline: apt-get -y install msbuild
Install: msbuild-sdkresolver:amd64 (15.3+xamarinxplat.2017.07.20.13.52-0xamarin3+ubuntu1404b1, automatic), msbuild-libhostfxr:amd64 (2.0.0.2017.07.06.00.01-0xamarin14+debian7b1, automatic)
Upgrade: msbuild:amd64 (15.2+xamarinxplat.2017.05.04.00.44-0xamarin3+ubuntu1404b1, 15.3+xamarinxplat.2017.07.20.13.52-0xamarin3+ubuntu1404b1)
End-Date: 2017-08-27  18:53:51`

But I can't find any relation with plexdrive. 
Anyway I have removed it as not needed anymore.

**Comment by kuarthepirat** [comment]
@rafuz yup, still stable. It does not seem to be the installed software alone however. I tried to replicate the issue on my system at home (current packages in ArchLinux) and I couldn't.

My current hypothesis (based on guessing) is, that it could be the garbage collection (because plexdrive consumes much more ram than it should based on MaxChunks). Once you reach a certain threshold in RAM the download of a new chunk is blocked and that locks the whole drive. This happens faster when you have less RAM or more users watching stuff on your server.

Maybe a frequent restart of plexdrive, like every 3 hours, can ease that issue?

**Comment by AndreAuth** [comment]
Still not getting better. Almost every day a freeze

**Comment by rafuz** [comment]
Removing JD2 and Ombi seems to improve the situation. They both use a lot of memory on the system. Mybe the guessing of @kuarthepirat could be realistic.
I am running for 4 days now and it is still stable.

**Comment by AndreAuth** [comment]
I am not quite sure. I currently have 26 GB memory free. Also i dont have jd2 and ombi installed on my system.

**Comment by djsecrist** [comment]
Have you considered that it might be something within your GD content? Maybe duplicates or something like that? Have you tried a different Google Drive with your same system? Or remove dups (https://rclone.org/commands/rclone_dedupe/) and set up a fresh PlexDrive config? Like I said before, I have been running PD5 for months without ANY issues on a system with 8GB of ram, 5k movies, and usually 2-3 streams while in use.

**Comment by rafuz** [comment]
I have no duplicates, my gdrive is well organized. I have deleted the configuration and redone it a couple of times, I have even disabled and reenabled gdrive api.

**Comment by Soulplayer** [comment]
I'm having the exact same problem. It's driving me insane... It worked perfectly until a couple of weeks ago then random freezes. Tomorrow I'm going to try to roll back the server and packages. 

I'm running rutorrent, sickrage, couchpotato, autosub and plex.

**Comment by rafuz** [comment]
@kuarthepirat Hello there, are you still running fine? :-)

I had to restart mine after about 10 days running fine. 

But this time it had a slightly different problem. It was not completely stuck but still downloaded some chunks, only incredibly slow (about 700 kbits instead of the usual 50 mbits). Only the "preloaded" chunks where really slow, the first chunk downloaded was full-speed. So when starting playing something I had a spike to 50-60 mbit for the first chunk and then slowed down for preloads to a speed not useful for fullhd content.
Nevertheless restarting plexdrive process fixed the slowness instantly.

Thoughts about that?

**Comment by kuarthepirat** [comment]
@rafuz that is funny, I had the exact same problem yesterday. And exactly as you described a quick restart (without having to kill the plexdrive process) fixed the problem.

**Comment by rafuz** [comment]
Looking at Ajkis scripts here [https://github.com/ajkis/scripts/tree/master/plexdrive](url) I have found that he mounts plexdrive using a lot of threads (20) for --chunk-check-threads and --chunk-load-threads
I am trying this settings right now.

**Comment by kuarthepirat** [comment]
I had this problem only once, I think that may have been some sort of connection issue.

**Comment by flpotro** [comment]
Hello,

I have this problem too. It locks randomly... 

What Plexdrive configuration do you have?

This is mine:

`/usr/bin/plexdrive mount -c /etc/plexdrive -o allow_other /media/plexlibrary
`

through `/etc/systemd/system` service...

**Comment by rafuz** [comment]
I also user a systemd service, my running config is in this thread but I don't think that are related to the issue anyway.
Yesterday in the evening I had another slow-down like the last time. Restarting plexdrive had no results this time. I had to completely reboot the server to get back to normal transfer speeds.

**Comment by flpotro** [comment]
Yesterday I had also another slow-down! and it got back to normal just rebooting! My hoster is online.net ... maybe the hoster is the problem?

**Comment by WillMonaghan** [comment]
I've been been following this issue, i haven't had a lock up for a couple of weeks but have experienced them a few times. Everything looks like it is still connected but nothing plays, like @flpotro i'm with online.net

**Comment by AndreAuth** [comment]
As mentioned i am with scaleway

**Comment by flpotro** [comment]
Scaleway=online.net ... Interesting...

**Comment by SeppGeneral** [comment]
Hmm my is OneProvider=Online.net
Mystery

**Comment by flpotro** [comment]
No locks today...

**Comment by AndreAuth** [comment]
Hmm @rafuz is at OVH. Is this the same Datacenter as Online.net ?

**Comment by rafuz** [comment]
I am in OVH RBX datacenter but it is really strange that we are experiencing this issue nearly at the same time.

**Comment by flpotro** [comment]
Very strange... maybe Google Drive issue with France location?

**Comment by fraviktor** [comment]
My servers' locations are in Paris and in Amsterdam. If I remember correctly, the problem only occured on the server located at Paris. The location can be the origin of the issue...
BUT:
I have no issues lately, the plexdrive service is up since 2017-11-14 without any problems.

**Comment by rafuz** [comment]
It is very interesting that could be related to a connection issue from Google from a specific localtion. @kuarthepirat are you hosting in France like us?

**Comment by AndreAuth** [comment]
My Server is also located in Paris

**Comment by Soulplayer** [comment]
I did a clean reinstall without updating the kernel and the packages and everything is running great since 25/11. I'm also located in Paris. I've had no slowdowns in this period.

**Comment by flpotro** [comment]
I have my OS updated...

**Comment by bassrock** [comment]
I have seen this now as well. I have added extra logging to a remote server to try and find the cause. My server is located located in the bay area. I am running FreeNAS 11.

**Comment by rafuz** [comment]
I still can't track the cause of this issue.
My plexdrive is running fine for 5 days now... I dont know when it will happen again.

**Comment by NeoPhyTe-x360** [comment]
Same here. Clean install of debian, latest plexdrive. Plexdrive is not downloading new chunks

**Comment by kuarthepirat** [comment]
I have switched my provider in the beginning of december and am totally free of any plexdrive problems so far. I am pretty convinced now that it is somehow related to the datacenter or the provider and my hypothesis is, that the requests are not satisfied by google drive and since there is no timeout function in plexdrive the threads block and sooner or later as more download threads block, the drive becomes unresponsive. 

Do you all still have your problems? Maybe we could add some sort of timeout to the chunk downloader.

**Comment by dweidenfeld** [comment]
Good idea. I will mention it in a future version.

**Comment by rafuz** [comment]
I think that @kuarthepirat is right about this issue. I had some connection problems to gsuite API from my server about three weeks ago (bad peering and limited bandwidht). I asked for support to the provider and it seems to be finally fixed. Using more threads obviously increased the time needed to freeze plexdrive but now I am running with just 8 threads.

**Comment by neckaros** [comment]
same issue here. sometime i have to restart my plexdrive 4times in a movie.
my server is not on online.net but local in Paris so it may indeed be location issue.
i hope a timeout management will fix this 😁
